src/lib/hooks/use-current-user-role.ts

```diff
@@ -22,7 +23,7 @@ export const useCurrentUserRole = () => {
     
     return {
       // Current relationship-specific role (for presence indicators, session UI)
-      role: isCoachInCurrentRelationship ? 'coach' as const : 'coachee' as const,
+      relationship_role: (isCoachInCurrentRelationship ? 'coach' : 'coachee') as RelationshipRole,
```

Comment:
I'd prefer that we use `RelationshipRole.Coach` or `RelationshipRole.Coachee` and define the type as such instead of a string union type. This should give us better and stricter type checks as the main advantage.

Reply:
Excellent suggestion 711bc7f2dfc5023f55da6ee37f6c82c3c3d2590d